Python/Product/Cookiecutter/Model/GitClient.cs

```diff
@@ -52,8 +53,13 @@ class GitClient : IGitClient {
             }
 
             var arguments = new string[] { "clone", repoUrl };
-            if (!ExistsOnPath(_gitExeFilePath)) {
-                throw new FileNotFoundException();
+            try {
+                // check if it's actually git
+                if(!canRunGit(_gitExeFilePath)) {
+                    throw new GitRunException(Strings.GitFailedToRunError);
+                }
+            } catch (Exception) {
+                throw new GitRunException(Strings.GitFailedToRunError);
```

Comment:
This should include the original exception as the inner exception for the one that it throws. This way, if it escapes and gets reported in a log, the full context is available.